src/Plugin/rest/resource/WebformAllFormSubmissions.php

```diff
@@ -106,53 +109,91 @@ public function get(string $webform_id): ModifiedResourceResponse {
-    if (is_numeric($startTimestamp)) {
-      $query->condition('created', $startTimestamp, '>=');
-      $result['starttime'] = $startTimestamp;
-    }
+    foreach (self::ALLOWED_DATETIME_QUERY_PARAMS as $param => $operator) {
+      $errors = $this->updateSubmissionQuery($this->currentRequest, $submissionQuery, $param, $operator, $result);
 
-    $endTimestamp = $this->currentRequest->query->get('endtime');
-    if (is_numeric($endTimestamp)) {
-      $query->condition('created', $endTimestamp, '<=');
-      $result['endtime'] = $endTimestamp;
+      if (isset($errors['error'])) {
+        return new ModifiedResourceResponse($errors, Response::HTTP_BAD_REQUEST);
+      }
     }
 
-    $query->accessCheck(FALSE);
-    $sids = $query->execute();
+    // Complete query.
+    $submissionQuery->accessCheck(FALSE);
+    $sids = $submissionQuery->execute();
 
     $submissionData = [];
 
-    foreach ($sids as $sid) {
-      /** @var \Drupal\webform\WebformSubmissionInterface $webform_submission */
-      $webform_submission = $this->entityTypeManager->getStorage('webform_submission')->load($sid);
-
-      $url = Url::fromRoute(
-            'rest.webform_rest_submission.GET',
-            [
-              'webform_id' => $webform_id,
-              'uuid' => $webform_submission->uuid(),
-            ],
-            [
-              'absolute' => TRUE,
-            ]
-        )->toString();
-
-      $submissionData[$sid] = $url;
+    if (!empty($sids)) {
+      $submissions = $submissionEntityStorage->loadMultiple($sids);
+
+      foreach ($submissions as $submission) {
+        $url = Url::fromRoute(
+          'rest.webform_rest_submission.GET',
+          [
+            'webform_id' => $webform_id,
+            'uuid' => $submission->uuid(),
+          ]
+        )
+          ->setAbsolute()
+          ->toString(TRUE)->getGeneratedUrl();
+
+        $submissionData[$submission->id()] = $url;
+      }
     }
 
     $result['submissions'] = $submissionData;
 
     return new ModifiedResourceResponse($result);
   }
 
+  /**
+   * Updates submission query with request query parameters.
+   *
+   * @phpstan-param array<string, mixed> $result
+   * @phpstan-return array<string, mixed>
+   */
+  private function updateSubmissionQuery(Request $request, QueryInterface $submissionQuery, string $parameter, string $operator, array &$result): array {
+    // Handle starttime request query.
+    $timeQuery = $request->query->get($parameter);
+
+    if (!empty($timeQuery)) {
+      try {
+        $startTime = new \DateTime($timeQuery);
+        $submissionQuery->condition('created', $startTime->getTimestamp(), $operator);
+        $result[$parameter] = $timeQuery;
+      }
+      catch (\Exception $e) {
+        $errors = [
+          'error' => [
+            'message' => $this->t('Could not generate DateTime from :time', [':time' => $timeQuery]),
+          ],
+        ];
+      }
+    }
+
+    return $errors ?? [];
+  }
```

Comment:
Having this function both return errors (if any) and adding data to `$result` makes it really convoluted.

Move the function body inside the `foreach` loop in line 134 to make the code easier to understand.

Reply:
Agreed. Fixed.